docs/devguide.md

```diff
@@ -297,6 +298,36 @@ If you choose third party resources storage, the helm chart will not launch an
 etcd server, but will instead instruct the API server to store all resources in
 the Kubernetes cluster as third party resources.
 
+## Dependency Management
+We use [dep](https://golang.github.io/dep) manage our dependencies. We commit the resulting
+vendor directory to ensure repeatable builds and isolation from upstream source disruptions.
+Because vendor is committed, you do not need to interact with dep unless you are
+changing dependencies.
+
+* Gopkg.toml - the dep manifest, this is intended to be hand-edited and contains a set of
+constraints and other rules for dep to apply when selecting appropriate versions of dependencies.
+* Gopkg.lock - the dep lockfile (generated). For the most part, don't edit this file because dep
+completely overwrites that file during `dep ensure`. If you run into trouble with this file, ask.
+* vendor/ - the source of all of our dependencies. Commit changes to this directory in a 
+separate commit from any other changes (including to the Gopkg files) so that it's easier to 
+review your pull request.
+
+If you use VS Code, we recommend installing the [dep extension](https://marketplace.visualstudio.com/items?itemName=carolynvs.dep).
+It provides snippets and improved highlighting that makes it easier to work with dep.
+
+### Add a new dependency
+1. Run `dep ensure -add github.com/example/project/pkg/foo`. This adds a constraint to Gopkg.toml,
+and downloads the dependency to vendor/.
+1. Import the package in the code and use it.
+1. Run `dep ensure -v` to sync Gopkg.lock and vendor/ with your changes.
+
+### Change the version of a dependency
```

Comment:
Do we have any guidelines on dependency versioning, e.g. is master ok? or do we require major/minor/strict version?

Reply:
Dep discourages (but doesn't restrict) using a specific revision in the Gopkg.toml file. You can track master (usually for projects that don't version their releases), another branch, or a specific non-semver tag, semver range, etc. I can add a comment about preferring a semver when available, and when not using master instead of a specific revision. If a package is pinned, it should probably be accompanied with a comment explaining why.

I'll update the PR with that.